tests/subtractions/test_api.py

```diff
@@ -144,268 +112,344 @@ async def test_get_from_job(fake: DataFaker, spawn_job_client, snapshot_recent):
-            },
+        upload = await fake.uploads.create(
+            user=user,
+            upload_type=UploadType.subtraction,
+            name="foobar.fq.gz",
+        )
+        subtraction = await fake.subtractions.create(
+            user=user,
+            upload=upload,
+            finalized=True,
         )
 
-    resp = await client.delete("subtractions/foo")
+        resp = await client.delete(f"subtractions/{subtraction.id}")
 
-    assert resp.status == 204 if exists else 400
+        assert resp.status == HTTPStatus.NO_CONTENT
+    else:
+        resp = await client.delete("subtractions/NOT_FOUND_ID")
+        assert resp.status == HTTPStatus.NOT_FOUND
 
 
-@pytest.mark.parametrize("error", [None, "404_name", "404", "409"])
-async def test_upload(
-    error: str | None,
-    pg: AsyncEngine,
-    resp_is,
-    mongo: Mongo,
-    spawn_job_client,
-    snapshot,
-    tmp_path: Path,
-):
-    client = await spawn_job_client(authenticated=True)
-    test_dir = tmp_path / "files"
-    test_dir.mkdir()
-    test_dir.joinpath("subtraction.1.bt2").write_text("Bowtie2 file")
+class TestUploadSubtractionFile:
+    @pytest.fixture(autouse=True)
+    async def setup(
+        self,
+        fake: DataFaker,
+        tmp_path: Path,
+    ):
+        test_dir = tmp_path / "files"
+        test_dir.mkdir(exist_ok=True)
 
-    get_config_from_app(client.app).data_path = tmp_path
+        self.test_file_directory: Path = test_dir
+        self.test_subtraction_file_directory: Path = tmp_path / "subtractions"
 
-    if error == "409":
-        async with AsyncSession(pg) as session:
-            session.add(SQLSubtractionFile(name="subtraction.1.bt2", subtraction="foo"))
-            await session.commit()
+        self.VALID_SUBTRACTION_FILE_NAME = "subtraction.1.bt2"
+        self.INVALID_SUBTRACTION_FILE_NAME = "reference.1.bt2"
 
-    if error != "404":
-        await mongo.subtraction.insert_one({"_id": "foo", "name": "Foo"})
+        user = await fake.users.create()
+        upload = await fake.uploads.create(
+            user=user,
+            upload_type=UploadType.subtraction,
+            name="foobar.fq.gz",
+        )
 
-    url = "/subtractions/foo/files"
+        self.subtraction = await fake.subtractions.create(user=user, upload=upload)
 
-    if error == "404_name":
-        url = f"{url}/reference.1.bt2"
-    else:
-        url = f"{url}/subtraction.1.bt2"
+        async def teardown():
+            self.test_file_directory.joinpath(self.VALID_SUBTRACTION_FILE_NAME).unlink()
 
-    resp = await client.put(
-        url,
-        data={"file": open(test_dir / "subtraction.1.bt2", "rb")},
-    )
+        return teardown
+
+    async def test_not_found(
+        self,
+        spawn_job_client: JobClientSpawner,
+        resp_is,
+    ):
+        client = await spawn_job_client(authenticated=True)
+
+        resp = await client.put(
+            f"/subtractions/TEST_MISSING_ID/files/{self.VALID_SUBTRACTION_FILE_NAME}",
```

Comment:
I would change `TEST_MISSING_ID` to something like `does_not_exist` for clarity.

Since you are using an f-string here, my first thought was that you were referring to a constant and forgot the `{}`. All-caps are generally reserved for constant variables.